asdf/extension/_legacy.py

```diff
@@ -186,65 +184,27 @@ def url_mapping(self):
 
 
 class _DefaultExtensions:
-    def __init__(self):
-        self._extensions = []
-        self._extension_list = None
-        self._package_metadata = {}
-
-    def _load_installed_extensions(self, group='asdf_extensions'):
-        for entry_point in iter_entry_points(group=group):
-            with warnings.catch_warnings(record=True) as w:
-                warnings.simplefilter('always', category=AsdfDeprecationWarning)
-                ext = entry_point.load()
-            if not issubclass(ext, AsdfExtension):
-                warnings.warn("Found entry point {}, from {} but it is not a "
-                              "subclass of AsdfExtension, as expected. It is "
-                              "being ignored.".format(ext, entry_point.dist),
-                              AsdfWarning)
-                continue
-
-            dist = entry_point.dist
-            name = get_class_name(ext, instance=False)
-            self._package_metadata[name] = (dist.project_name, dist.version)
-            self._extensions.append(ext())
-
-            for warning in w:
-                warnings.warn('{} (from {})'.format(warning.message, name),
-                              AsdfDeprecationWarning)
-
     @property
     def extensions(self):
-        # This helps avoid a circular dependency with external packages
-        if not self._extensions:
-            # If this environment variable is defined, load the default
-            # extension. This allows the package to be tested without being
-            # installed (e.g. for builds on Debian).
-            if os.environ.get(ASDF_TEST_BUILD_ENV):
-                # Fake the extension metadata
-                name = get_class_name(BuiltinExtension, instance=False)
-                self._package_metadata[name] = ('asdf', asdf_version)
-                self._extensions.append(BuiltinExtension())
-
-            self._load_installed_extensions()
-
-        return self._extensions
+        from ..config import get_config
+        return [e for e in get_config().extensions if e.legacy]
 
     @property
     def extension_list(self):
-        if self._extension_list is None:
-            self._extension_list = AsdfExtensionList(self.extensions)
-
-        return self._extension_list
+        return AsdfExtensionList(self.extensions)
```

Comment:
This is exceedingly too clever for me. This class(?) property uses self.extensions, but the class has no __init__ nor inherits from another class, and yet has that attribute? I need to catch up on my meta meta classes.

Reply:
The extensions property is still there!  It's just obscured on GitHub by the ugly diff.

https://github.com/asdf-format/asdf/blob/c900f3fd2e3fd472df37f2ba21d60b9a29c571b1/asdf/extension/_legacy.py#L188

Reply:
Indeed. Sorry...